### PR TITLE
[IMPAC-263] Added hist mode to custom calculation (9)

### PR DESCRIPTION
**Pull request by an-pavlovych**
In this task I've added `hist mode choser` as directive to `accounts-custom-calculation.tmpl.html`. Also added `setting-time-period` directive to template with widget settings. Though I faced the problem: these two directives some how create different `hist_parameters` objects in `widget.settings`. Hist mode switcher goes broken because of it. I.e. `hist mode` works till we open settings. Then `setting-time-period` addes new object `hist_parameters` to `widget.settings`, and do not use created. For now I cannot find the reason of this bug.

### CODE REVIEW

File: src/components/widgets-settings/accounts-list/accounts-list.directive.coffee [hunk 1]
```diff
@@ -33,9 +33,8 @@ module.controller('SettingAccountsListCtrl', ($scope, $timeout, ImpacWidgetsSvc)
 
     if w.content? && !_.isEmpty(w.content.complete_list)
       w.remainingAccounts = angular.copy(w.content.complete_list)
-      $timeout () ->
```
Comment (cesar-tonnoir) [inline src/components/widgets-settings/accounts-list/accounts-list.directive.coffee]: @an-pavlovych the timeout was there to ensure all the context is passed to the directive before executing the next statements.
Any particular reason for removing it?

File: src/components/widgets-settings/formula/formula.directive.coffee [hunk 2]
```diff
@@ -15,9 +15,8 @@ module.controller('SettingFormulaCtrl', ($scope, $filter, $timeout) ->
   setting.initialize = ->
     if w.metadata? && w.metadata.formula?
       w.formula = w.metadata.formula
-      $timeout () ->
```
Comment (cesar-tonnoir) [inline src/components/widgets-settings/formula/formula.directive.coffee]: @an-pavlovych same here ?

**Comment by an-pavlovych** [comment]
I've removed async there in order to have data calculated by formula in `WidgetAccountsCustomCalculationCtrl` directove in `w.format` method when creating the chart.

File: src/components/widgets-settings/formula/formula.directive.coffee [hunk 3]
```diff
@@ -65,13 +68,25 @@ module.controller('SettingFormulaCtrl', ($scope, $filter, $timeout) ->
...
+    return evaluatedFormula
+
   formatFormula = ->
     if !w.formula.match(/\//g) && w.selectedAccounts?
       if firstAcc = w.selectedAccounts[0]
         if currency = firstAcc.currency
           w.evaluatedFormula = $filter('mnoCurrency')(w.evaluatedFormula, currency)
 
-  w.settings.push(setting)
+  # w.settings.push(setting)
```
Comment (cesar-tonnoir) [inline src/components/widgets-settings/formula/formula.directive.coffee]: the widget will never load its content if you remove this line..

File: src/components/widgets-settings/formula/formula.directive.coffee [hunk 4]
```diff
@@ -48,15 +48,18 @@ module.controller('SettingFormulaCtrl', ($scope, $filter, $timeout) ->
...
-    if (!str.match(authorized_regex))
-      w.isFormulaCorrect = false
-      w.evaluatedFormula = "invalid expression"
+    w.evaluatedFormula = applyFormula(str)
 
-    try
-      w.evaluatedFormula = eval(str).toFixed(2)
-    catch e
-      w.evaluatedFormula = "invalid expression"
+    if w.selectedAccounts? && firstAcc = w.selectedAccounts[0]
```
Comment (cesar-tonnoir) [inline src/components/widgets-settings/formula/formula.directive.coffee]: @an-pavlovych code duplication here. The string replacement could have been put in a function and re-used

**Comment by cesar-tonnoir** [comment]
changes + corrections included in https://github.com/maestrano/impac-angular/pull/241

**Pull request closed without merging**